### PR TITLE
REM-75

### PR DESCRIPTION
**Pull request by RebeccaFrem02**
1- i can add a duplicated name

2- in new mode after save “info” button should be enable

3- in the second tab “interface” no need for save and clear button

### CODE REVIEW

File: src/components/Shared/InterfacesForm.js [hunk 1]
```diff
@@ -136,7 +136,14 @@ export const InterfacesForm = ({ recordId, expanded, height, resourceId, name })
   ]
```
Comment (omarZrayka23) [inline src/components/Shared/InterfacesForm.js:137]: onSubmit please add try and catch

File: src/pages/cb-banks/Windows/CbBanksWindow.js [hunk 2]
```diff
@@ -5,21 +5,28 @@ import CustomTabPanel from 'src/components/Shared/CustomTabPanel'
...
 
-const CbBanksWindow = ({ height, labels, maxAccess, recordId }) => {
+const CbBanksWindow = ({ labels, maxAccess, recordId, height }) => {
   const [activeTab, setActiveTab] = useState(0)
 
   const [store, setStore] = useState({
     recordId: recordId || null,
     name: null
   })
+  const [editMode, setEditMode] = useState(!!recordId)
```
Comment (omarZrayka23) [inline src/pages/cb-banks/Windows/CbBanksWindow.js:15]: can use edit mode as constant instead state

File: src/pages/cb-banks/forms/CbBanksForm.js [hunk 3]
```diff
@@ -38,29 +39,30 @@ export default function CbBanksForms({ labels, maxAccess, recordId, setStore })
...
     validateOnChange: true,
     validationSchema: yup.object({
-      name: yup.string().required(' '),
-      reference: yup.string().required(' '),
-      swiftCode: yup.string().required(' ')
+      name: yup.string().required(),
+      reference: yup.string().required(),
+      swiftCode: yup.string().required()
     }),
     onSubmit: async obj => {
```
Comment (omarZrayka23) [inline src/pages/cb-banks/forms/CbBanksForm.js:46]: also  please add try and catch

File: src/pages/cb-banks/forms/CbBanksForm.js [hunk 4]
```diff
@@ -38,29 +39,30 @@ export default function CbBanksForms({ labels, maxAccess, recordId, setStore })
...
       const response = await postRequest({
         extension: CashBankRepository.CbBank.set,
         record: JSON.stringify(obj)
       })
 
-      if (!obj.recordId) {
+      if (!formik.values.recordId) {
```
Comment (omarZrayka23) [inline src/pages/cb-banks/forms/CbBanksForm.js:52]: whu u instead obj.recordId

File: src/pages/cb-banks/forms/CbBanksForm.js [hunk 5]
```diff
@@ -38,29 +39,30 @@ export default function CbBanksForms({ labels, maxAccess, recordId, setStore })
     enableReinitialize: true,
     validateOnChange: true,
     validationSchema: yup.object({
-      name: yup.string().required(' '),
-      reference: yup.string().required(' '),
-      swiftCode: yup.string().required(' ')
+      name: yup.string().required(),
```
Comment (omarZrayka23) [inline src/pages/cb-banks/forms/CbBanksForm.js:42]: || null  removed in line 31

File: src/pages/cb-banks/Windows/CbBanksWindow.js (hunk 2 above)
Comment (RebeccaFrem02) [reply src/pages/cb-banks/Windows/CbBanksWindow.js:15]: no, i am using setEditMode in the form

File: src/pages/cb-banks/forms/CbBanksForm.js [hunk 6]
```diff
@@ -38,30 +39,34 @@ export default function CbBanksForms({ labels, maxAccess, recordId, setStore })
...
-      } else {
-        toast.success(platformLabels.Edited)
-      }
-      setStore({
-        recordId: response.recordId,
-        name: obj.name
-      })
-      invalidate()
+
+        if (!formik.values.recordId) {
```
Comment (omarZrayka23) [inline src/pages/cb-banks/forms/CbBanksForm.js:53]: can u use obj.recordId

File: src/pages/cb-banks/Windows/CbBanksWindow.js (hunk 2 above)
Comment (omarZrayka23) [reply src/pages/cb-banks/Windows/CbBanksWindow.js:15]: why no? we stoped using setEditMode

File: src/pages/cb-banks/index.js [hunk 7]
```diff
@@ -11,9 +11,11 @@ import { useWindow } from 'src/windows'
...
 import { Grow } from 'src/components/Shared/Layouts/Grow'
 import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { ControlContext } from 'src/providers/ControlContext'
 
 const CbBank = () => {
   const { getRequest, postRequest } = useContext(RequestsContext)
+  const { platformLabels } = useContext(ControlContext)
   const { stack } = useWindow()
 
   async function fetchGridData(options = {}) {
```
Comment (omarZrayka23) [inline src/pages/cb-banks/index.js:21]: are u sure pagination is work?

**Comment by omarZrayka23** [comment]
please add  invalidate in useResourceQuery

**Comment by omarZrayka23** [comment]
in index.js ,        i do not need  recordId ? recordId : null

**Comment by omarZrayka23** [comment]
in index.js , i do not need recordId ? recordId : null

**Comment by omarZrayka23** [comment]
how   paginationType='client' and  api  .page

File: src/pages/cb-banks/Windows/CbBanksWindow.js [hunk 8]
```diff
@@ -5,7 +5,7 @@ import CustomTabPanel from 'src/components/Shared/CustomTabPanel'
 import { InterfacesForm } from 'src/components/Shared/InterfacesForm'
 import { ResourceIds } from 'src/resources/ResourceIds'
 
-const CbBanksWindow = ({ height, labels, maxAccess, recordId }) => {
+const CbBanksWindow = ({ labels, maxAccess, recordId, height }) => {
   const [activeTab, setActiveTab] = useState(0)
 
   const [store, setStore] = useState({
```
Comment (omarZrayka23) [inline src/pages/cb-banks/Windows/CbBanksWindow.js:11]: remove || null

File: src/pages/cb-banks/Windows/CbBanksWindow.js [hunk 9]
```diff
@@ -19,7 +19,7 @@ const CbBanksWindow = ({ height, labels, maxAccess, recordId }) => {
     <>
       <CustomTabs tabs={tabs} activeTab={activeTab} setActiveTab={setActiveTab} />
       <CustomTabPanel height={height} index={0} value={activeTab}>
-        <CbBanksForm labels={labels} maxAccess={maxAccess} recordId={recordId} setStore={setStore} />
+        <CbBanksForm labels={labels} maxAccess={maxAccess} store={store} setStore={setStore} />
```
Comment (omarZrayka23) [inline src/pages/cb-banks/Windows/CbBanksWindow.js:22]: remove height={height}

File: src/components/Shared/InterfacesForm.js [hunk 10]
```diff
@@ -39,27 +39,32 @@ export const InterfacesForm = ({ recordId, resourceId, name }) => {
     enableReinitialize: true,
```
Comment (omarZrayka23) [inline src/components/Shared/InterfacesForm.js:39]: can u clear import { Fixed } from './Layouts/Fixed'

File: src/components/Shared/InterfacesForm.js [hunk 11]
```diff
@@ -39,27 +38,32 @@ export const InterfacesForm = ({ recordId, resourceId, name }) => {
...
-      const resultRows = hasEmptyRows ? [] : rows
+        const resultRows = hasEmptyRows ? [] : rows
 
-      const data = {
-        recordId: recordId,
-        resourceId: resourceId,
-        items: resultRows
-      }
+        const data = {
+          recordId: recordId,
```
Comment (omarZrayka23) [inline src/components/Shared/InterfacesForm.js:53]: u can use recordId, direct

File: src/pages/cb-banks/forms/CbBanksForm.js [hunk 12]
```diff
@@ -38,32 +39,37 @@ export default function CbBanksForms({ labels, maxAccess, recordId, setStore })
...
-      }
-      setStore({
-        recordId: response.recordId,
-        name: obj.name
-      })
-      invalidate()
+
+        if (!obj.recordId) {
+          toast.success(platformLabels.Added)
+          formik.setValues({
```
Comment (omarZrayka23) [inline src/pages/cb-banks/forms/CbBanksForm.js:55]: formik.setFieldValue

**Pull request merged**